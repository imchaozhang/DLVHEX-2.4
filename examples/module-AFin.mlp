#module(p1,[]).
q1(a).
q1(b).
r1(a) :- @p3[q1]::c. 
r1(b) :- r1(a). 
ok1 :- @p2[r1]::ok2.
bad1 :- not ok1.

#module(p2,[q2/1]).
ok2 :- @p3[q2]::c.
bad2 :- not ok2.

#module(p3,[q3/1]).
c :- q3(a), q3(b).

