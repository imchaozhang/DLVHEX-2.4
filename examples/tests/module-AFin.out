(p1[{}]={q1(a),q1(b),r1(a),r1(b),ok1}, p3[{q3(a),q3(b)}]={q3(a),q3(b),c}, p2[{q2(a),q2(b)}]={q2(a),q2(b),ok2})
